server/lib/ldap/Manager.ts

```diff
@@ -0,0 +1,403 @@
+import limax from 'limax';
+import { SHA256 } from 'meteor/sha';
+import { Meteor } from 'meteor/meteor';
+import { Accounts } from 'meteor/accounts-base';
+import ldapEscape from 'ldap-escape';
+import _ from 'underscore';
+
+import { ILDAPEntry } from '../../../definition/ldap/ILDAPEntry';
+import { LDAPLoginResult } from '../../../definition/ldap/ILDAPLoginResult';
+import { ILDAPUniqueIdentifierField } from '../../../definition/ldap/ILDAPUniqueIdentifierField';
+import { IUser, /* IUserEmail,*/ LoginUsername } from '../../../definition/IUser';
+import { IImportUser } from '../../../app/importer/server/definitions/IImportUser';
+import { settings } from '../../../app/settings/server';
+import { Users as UsersRaw } from '../../../app/models/server/raw';
+import { LDAPConnection } from './Connection';
+import { LDAPDataConverter } from './DataConverter';
+import { getLDAPConditionalSetting } from './getLDAPConditionalSetting';
+import { logger, authLogger, connLogger } from './Logger';
+import type { IConverterOptions } from '../../../app/importer/server/classes/ImportDataConverter';
+import { callbacks } from '../../../app/callbacks/server';
+import { setUserAvatar } from '../../../app/lib/server/functions';
+
+export class LDAPManager {
+	public static async login(username: string, password: string): Promise<LDAPLoginResult> {
+		logger.log({ msg: 'Init LDAP login', username });
+
+		if (settings.get('LDAP_Enable') !== true) {
+			return this.fallbackToDefaultLogin(username, password);
+		}
+
+		const ldap = this.getNewConnection();
```

Comment:
does this mean for every new login attempt a new LDAP connection would be established? I didn't found if the connection is being closed afterwards, I guess we should?